sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/highlighting/XMLHighlighting.java

```diff
@@ -0,0 +1,216 @@
+
+    if (isEmptyElement) {
+      // empty (or autoclosing) element is raised twice as start and end element, so we need to highlight closing "/" which is placed just before ">"
+      highlighting.add(new HighlightingData(closingBracketStartOffset - 1, closingBracketStartOffset, "k"));
+
+    } else {
+      highlighting.add(new HighlightingData(startOffset, closingBracketStartOffset + 1, "k"));
+    }
+  }
+
+  private void highlightStartElement(XMLStreamReader xmlReader, int startOffset) {
+    int closingBracketStartOffset = getTagClosingBracketStartOffset(startOffset);
+    int endOffset = startOffset + getNameWithNamespaceLength(xmlReader) + 1;
+
+    highlighting.add(new HighlightingData(startOffset, endOffset, "k"));
+    highlightAttributes(endOffset, closingBracketStartOffset);
+    highlighting.add(new HighlightingData(closingBracketStartOffset, closingBracketStartOffset + 1, "k"));
+  }
+
+  private void highlightXmlDeclaration() {
+    if (content.startsWith(XML_DECLARATION_TAG)) {
+      int startOffset = 0;
+      int closingBracketStartOffset = getTagClosingBracketStartOffset(0);
+
+      highlighting.add(new HighlightingData(startOffset, startOffset + XML_DECLARATION_TAG.length(), "k"));
+      highlightAttributes(XML_DECLARATION_TAG.length(), closingBracketStartOffset);
+      highlighting.add(new HighlightingData(closingBracketStartOffset - 1, closingBracketStartOffset + 1, "k"));
+    }
+  }
+
+  private void highlightAttributes(int from, int to) {
+    int counter = from + 1;
+
+    Integer startOffset = null;
+    Character attributeValueQuote = null;
+
+    while (counter < to) {
+      char c = content.charAt(counter);
+
+      if (startOffset == null && !Character.isWhitespace(c)) {
+        startOffset = counter;
+      }
+
+
+      if (attributeValueQuote != null && attributeValueQuote == c) {
+        highlighting.add(new HighlightingData(startOffset, counter + 1, "s"));
+        counter++;
+        startOffset = null;
+        attributeValueQuote = null;
+      }
+
+      if (c == '=' && attributeValueQuote == null) {
+        highlighting.add(new HighlightingData(startOffset, counter, "c"));
+
+        do {
+          counter++;
+          c = content.charAt(counter);
+        } while (c != '\'' && c != '"');
+
+        startOffset = counter;
+        attributeValueQuote = c;
+      }
+
+
+      counter++;
+    }
+  }
+
+  private int getTagClosingBracketStartOffset(int startOffset) {
+    return getClosingBracketStartOffset(startOffset, false);
+  }
+
+  private int getCDATAClosingBracketStartOffset(int startOffset) {
+    return getClosingBracketStartOffset(startOffset, true);
+  }
+
+  private int getClosingBracketStartOffset(int startOffset, boolean isCDATA) {
+    int counter = startOffset + 1;
+    while (startOffset < content.length()) {
+      if (content.charAt(counter) == '>' && (!isCDATA || content.charAt(counter - 1) == ']')) {
```

Comment:
@vilchik-elena minor: this does not highlight correctly `<![CDATA[ aa]>bb  ]]>`.